src/hid/midi.h

```diff
@@ -253,6 +253,7 @@ class MidiHandler
                 {
                     // invalid message go back to start ;p
                     pstate_ = ParserEmpty;
+                    Parse(byte);
```

Comment:
This recursive approach has a risk of unwanted side effects:
If the second status byte in a row is received, we
1. reset the state and enter the `Parse()` function again
2. properly parse the status byte, because now we're in the right section of the `switch`
3. return from `Parse()` and get back here 
4. process the remaining parts of the initial call to `Parse()`

After step 3 we would already be done. Step 4 is only there because of the recursion and it could easily break the parser if someone adds additional code to the end of the `Parse()` function.

Instead, you could also add this to the beginning of the parser:
```cpp
void Parse(uint8_t byte)
{
    // reset parser when status byte is received
    if((byte & kStatusByteMask)) 
    {
        pstate_ = ParserEmpty;
    }
    switch(pstate_)
    {
        // ...
    }
}
```

Reply:
Good catch! One interesting thing to note is you have to ignore the supposed "status byte" when you're receiving sysex, since it might just be part of the sysex data being sent.

Caught that with the unit test. They're the best!